pkg/uhttp/wrapper.go

```diff
@@ -37,6 +42,52 @@ func WithJSONResponse(response interface{}) DoOption {
 	}
 }
 
+type ErrorResponse interface {
+	Message() string
+}
+
+func WithErrorResponse(resource ErrorResponse) DoOption {
+	return func(resp *http.Response) error {
+		if resp.StatusCode >= 300 {
```

Comment:
Should we consider 301/302 redirects to be errors? I think >= 400 is more accurate.

Also this code could early return if the conditional were reversed. eg:

```
if resp.StatusCode < 400 {
  return nil
}
```